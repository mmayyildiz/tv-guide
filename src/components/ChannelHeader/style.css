.tvguide .tvg-channel-list {
    position: relative;
    top: -1px;
    left: 0;
    display: inline-flex;
    flex-direction: column;
    flex-wrap: nowrap;
    justify-content: flex-start;
    width: 150px;
    margin: 0;
    padding: 0;
    border-top: 1px solid rgba(118,120,123,.51);
    border-right: 1px solid rgba(118,120,123,.51);
}

.tvguide .tvg-channel-list li {
    position: relative;
    width: 100%;
    height: 90px;
    overflow: hidden;
    line-height: 90px;
    text-align: center;
    -webkit-user-select: none;
    user-select: none;
    list-style: none;
}

.tvguide .tvg-channel-list li span {
    position: absolute;
    bottom: 10px;
    left: 10px;
    font-family: "Apercu",Arial,sans-serif;
    line-height: 1;
    text-align: right;
    opacity: .7;
    font-size: 10px;
}

.tvguide .tvg-channel-list li img {
    vertical-align: middle;
}

