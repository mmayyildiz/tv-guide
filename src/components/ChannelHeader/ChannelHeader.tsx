import React from 'react';
import { Size } from '../../constants/sizeConstants';

import ChannelLogo from '../ChannelLogo/ChannelLogo';
import './style.css';

interface ChannelHeaderProps {
    channels: string[];
}

function ChannelHeader({ channels }: ChannelHeaderProps) {

    return (
        <ul className="tvg-channel-list" >
            { channels.map((sid, index) =>
                <li key={index}>
                    <div>
                        <span>{index}</span>
                        <ChannelLogo sid={sid} size={Size["100PX"]} />
                    </div>
                </li>
            )
            }
        </ul>
    )

}

export default ChannelHeader;