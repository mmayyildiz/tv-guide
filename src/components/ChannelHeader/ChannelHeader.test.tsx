import React from 'react';
import { shallow } from 'enzyme';
import ChannelHeader from './ChannelHeader';

const props = {
    channels: ['1', '2', '3']
};

const mountedComponent = shallow(<ChannelHeader {...props} />);

describe('CalendarHeader', () => {

    it('should render with props', () => {
        expect(mountedComponent).toMatchSnapshot();
    });

    it('should render a list ', () => {
        const lis = mountedComponent.find('li');
        expect(lis.length).toBe(3);
    });

});
