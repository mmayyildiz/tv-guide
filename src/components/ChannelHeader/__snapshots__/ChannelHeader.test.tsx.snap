// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarHeader should render with props 1`] = `ShallowWrapper {}`;
